app/code/Magento/Checkout/Block/Checkout/AttributeMerger.php

```diff
@@ -326,6 +326,21 @@ protected function getDefaultValue($attributeCode)
                     return $this->getCustomer()->getLastname();
                 }
                 break;
+            case 'middlename':
+                if ($this->getCustomer()) {
+                    return $this->getCustomer()->getMiddlename();
+                }
+                break;
+            case 'prefix':
+                if ($this->getCustomer()) {
+                    return $this->getCustomer()->getPrefix();
```

Comment:
This PR is on middle name, I would suggest to skip other fields for the sake of testing simplicity.